src/lib/checkVersion.ts

```diff
@@ -96,29 +92,19 @@ export class CheckVersion {
     return result 
   }
 
-  async handlePackageJson(location: string, packageJson: {version?: string} | null = null, packageLockJson: any = null) {
-    const getFiles = new GetFiles(this.fs)
-    const filtered = await getFiles.getFiles(location)
+  async handlePackageJson(location: string) {
+const npmHandler = new NPMPackageHandler(this.fs)
+let versions = await npmHandler.scan(location)
   
-    //read and assign files
-    for (const file of filtered) {
-      const contents = await this.fs.readFile(location + file, 'utf8')
-      const jsonData = JSON.parse(contents)
-      if (file.indexOf('lock') != -1) {
-        packageJson = packageParser.parse(jsonData)
-      } else {
-        packageLockJson = packageParser.parse(jsonData)
-      }
-    }
-  
-    if (packageJson === null) {
-      //change to set failed
-      return
+
+
+    if (!(versions['packageJsonLock']) || !(versions['packageJson'])){
+      this.core.setFailed(`No versions found for package or package-lock.`)
     }
   
-    this.compareVersions(packageJson['version'] || '', packageLockJson['version'] || '')
+    this.compareVersions(versions['packageJson'] || '', versions['packageJsonLock'] || '')
   
-    return packageLockJson['version'] 
+    return versions['packageJsonLock'] 
```

Comment:
I'm wondering if the comparison logic should also move to the npm class as this doens't apply for cargo. Thoughts?